### ISSUE TITLE
maxWidth not working as expected

### ISSUE THREAD

**Issue by massimobio** [skydoves/Balloon#738]
**Please complete the following information:**
- Library Version 1.6.9
- Affected Device(s) all devices
 
**Describe the Bug:**

I'm trying to make a Balloon that fills the width of the screen in portrait mode but maxes out to a certain width in portrait or on tablets but I must be doing something wrong:
```Kotlin
        val balloon = Balloon.Builder(requireContext())
            .setTextResource(R.string.forums_message)
            .setTextColorResource(R.color.black)
            .setTextGravity(Gravity.START)
            .setTextSize(15f)
            .setMarginHorizontal(8)
            .setHeight(BalloonSizeSpec.WRAP)
            .setWidth(BalloonSizeSpec.WRAP)
            .setMaxWidth(500)
            .setIconDrawableResource(R.drawable.ic_forums)
            .setArrowPositionRules(ArrowPositionRules.ALIGN_ANCHOR)
            .setArrowSize(20)
            .setPadding(8)
            .setCornerRadius(8f)
            .setBackgroundColorResource(R.color.popupBackground)
            .setBalloonAnimation(BalloonAnimation.ELASTIC)
            .setLifecycleOwner(viewLifecycleOwner)
            .build()
        forumsButton.showAlignBottom(balloon)
```
It works as expected in landscape:
![311953534228668408](https://github.com/user-attachments/assets/2860542d-0a67-47ce-aa08-2e18b54eb6b9)

but in portrait is forces the width of the balloon to the maxWidth(500) which makes it overflow the width of the screen:
![1118857044101946786](https://github.com/user-attachments/assets/ff4a92b2-3228-4960-bfbb-08ecebe6781d)

**Expected Behavior:**
I would expect that the content view's width would override `.setMaxWidth(500)`
If  remove `.setMaxWidth(500)` it looks as expected in portrait:
![1505275742964764086](https://github.com/user-attachments/assets/cf6cf60c-1e35-49c5-9560-ade09cc9cb04)

**Comment by skydoves** [comment]
Hey @massimobio, the `maxWidth` method lets the inner content measure itself based on the provided value, so this is actually the expected behavior. What about using the `maxWidth` method only in the landscape mode?

**Comment by massimobio** [comment]
Hi @skydoves, thank you for the clarification. 
Yes, I can manually set `maxWidth` on the Balloon if the screen width exceeds a certain amount.
As you mentioned, `maxWidth` relates to the content measure, but why disregard the width of the screen? 
When I see the method called `maxWidth`, I assume that I'm setting the upper limit for the balloon width but it can be smaller if the text is short or the available screen space is smaller. It seems rare that one would want a balloon to be bigger than the available screen size, but if one does, there is `minWidth`. Also, the other APIs like `maxWidthRatio` refer to the screen width and if I make a simple balloon with default values it adjusts to the screen width, so it seems like in some cases the screen width is the limiting factor, not the content measure.
Anyway, this is just my opinionated feedback 😊
Thank you for a great library and all other ones you develop and maintain 🙏

**Comment by skydoves** [comment]
Hey @massimobio, thank you so much for your valuable feedback! That makes a lot of sense. I fixed it and has been released on the recent version 1.6.10. Thank you for your report!
